diff --git a/rl_coach/dashboard.py b/rl_coach/dashboard.py
index da8eb1a..0acfb8f 100644
--- a/rl_coach/dashboard.py
+++ b/rl_coach/dashboard.py
@@ -43,6 +43,10 @@ parser.add_argument('-f', '--experiment_files',
                     help="(string) The path of an experiment file to open",
                     default=None,
                     type=str)
+parser.add_argument('-p', '--port',
+                    help="(string) Port of dashboard",
+                    default=None,
+                    type=str)
 args = parser.parse_args()
 
 if args.experiment_dir:
@@ -56,9 +60,12 @@ elif args.experiment_files:
 
 def main():
     from rl_coach.utils import get_open_port
+    port = args.port
+    if port == None:
+        port = get_open_port()
 
     dashboard_path = os.path.realpath(__file__)
-    command = 'bokeh serve --show {} --port {}'.format(dashboard_path, get_open_port())
+    command = 'bokeh serve --show {} --port {}'.format(dashboard_path, port)
     if args.experiment_dir or args.experiment_files:
         command += ' --args'
         if args.experiment_dir:
