diff --git a/coach/docker/1.0.0/Dockerfile.tf b/coach/docker/1.0.0/Dockerfile.tf
index 51d61ce..ef1793e 100644
--- a/coach/docker/1.0.0/Dockerfile.tf
+++ b/coach/docker/1.0.0/Dockerfile.tf
@@ -1,11 +1,11 @@
-ARG processor
-ARG region
-FROM 520713654638.dkr.ecr.$region.amazonaws.com/sagemaker-tensorflow-scriptmode:1.12.0-$processor-py3
+FROM local/sagemaker-tensorflow-container:nvidia
+
+ARG sagemaker_containers
 
 RUN apt-get update && apt-get install -y --no-install-recommends \
         build-essential \
         jq \
-        libav-tools \
+        ffmpeg \
         libjpeg-dev \
         libxrender1 \
         python3.6-dev \
@@ -28,9 +28,13 @@ RUN pip install --no-cache-dir \
     pyglet==1.3.2 \
     gym==0.12.5 \
     redis==2.10.6 \
-    rl-coach-slim==1.0.0 && \
-    pip install --no-cache-dir --upgrade sagemaker-containers && \
+    tensorflow==1.15.0 \
+    rl-coach-slim==1.0.0 
+
+COPY $sagemaker_containers .
+RUN pip install --no-cache-dir --upgrade "$sagemaker_containers" && \
     pip install --upgrade numpy
+RUN rm -f $sagemaker_containers
 
 ENV COACH_BACKEND=tensorflow
 
